src/librustc_metadata/decoder.rs

```diff
@@ -127,15 +128,15 @@ impl<'a, 'tcx> Metadata<'a, 'tcx> for (&'a CrateMetadata, TyCtxt<'a, 'tcx, 'tcx>
     }
 }
 
-impl<'a, 'tcx: 'a, T: Decodable> Lazy<T> {
+impl<'a, 'tcx: 'a, T: Encodable + Decodable> Lazy<T> {
```

Comment:
Where does these encodable bounds come from?

Reply:
The whole "lazy metadata" stuff, I was trying to make blanket impls not conflict with "more specialized" impls.